library/entropy.c

```diff
@@ -576,10 +578,12 @@ int mbedtls_entropy_source_self_test( int verbose )
 int mbedtls_entropy_self_test( int verbose )
 {
     int ret = 1;
+#if !defined(MBEDTLS_TEST_NULL_ENTROPY)
```

Comment:
I do not think so because you put these lines after the if statement i.e. somewhere around line 590.